src/install/index.ts

```diff
@@ -417,16 +417,18 @@ export class Installer {
       // auto-generate name from target if necessary
       if (!install.name) {
         if (typeof install.target !== 'string') {
-          if (install.target.name.substr(install.target.name.lastIndexOf(':') + 1).match(validPkgNameRegEx))
-            install.name = install.target.name.substr(install.target.name.lastIndexOf(':') + 1);
+          const idx = install.target.name.lastIndexOf(':') + 1;
+          const substr = install.target.name.substr(idx);
+          if (substr.match(validAlaisRegEx))
```

Comment:
I'm not sure I follow this part since `install.target` is supposed to be the `registry:name@version` thing right? So in `jspm install @x=x`, the `install.target` should be the `npm:x` part right? Not the `@x` part? It's been a while since I've looked at this code, so maybe you can clarify this for me?

Reply:
This was not really necessary for this PR.  This was a refactor to reduce some redundant code while I was debugging.